server/src/main/java/org/opensearch/env/NodeRepurposeCommand.java

```diff
@@ -94,12 +97,20 @@ protected boolean validateBeforeLock(Terminal terminal, Environment env) {
     @Override
     protected void processNodePaths(Terminal terminal, Path[] dataPaths, int nodeLockId, OptionSet options, Environment env)
         throws IOException {
-        assert DiscoveryNode.isDataNode(env.settings()) == false;
+        assert DiscoveryNode.isDataNode(env.settings()) == false || DiscoveryNode.isSearchNode(env.settings()) == false;
 
-        if (DiscoveryNode.isClusterManagerNode(env.settings()) == false) {
-            processNoClusterManagerNoDataNode(terminal, dataPaths, env);
-        } else {
-            processClusterManagerNoDataNode(terminal, dataPaths, env);
+        if (DiscoveryNode.isDataNode(env.settings()) == false) {
```

Comment:
What if both `isDataNode` and `isSearchNode` are false? This will only execute the first branch of the if statement. Is that correct?

Reply:
The idea here was that it would operate in a step fashion, because the errors are thrown individually - https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/env/NodeEnvironment.java#L386-L396

The clients can perform the repurposing one at a time. 
WDYT?

Reply:
So lets say I have a node that has been used as both a `data` and a `search` node and I want to repurpose it to be a cluster manager node. Does that mean I'd have to run the repurpose command twice?

Reply:
Yep, since the conditions are currently laid out differently within node startup.

Reply:
But node startup doesn't have anything to do with this repurpose command though, right? If I run this command on a host that had multiple roles but now no longer has any of them, I would expect it to do the right thing and clear everything out. Is that difficult or not possible for some reason?

Reply:
It's not difficult, just a bunch of combinatorial logic, which I was trying to avoid. I can add it in, shouldn't take a lot of time.

Reply:
Does it have to be combinatorial? Like why can't it be:

```
if not data role:
  clean up data if exists
if not search role:
  clean up search if exists
...
```

Reply:
Updated.